client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -153,49 +172,29 @@ export class SiteVerticalsSuggestionSearch extends Component {
 		);
 	}
 }
+export function requestVerticals( search, limit = 7, callback ) {
+	request
+		.get( 'https://public-api.wordpress.com/wpcom/v2/verticals' )
```

Comment:
I can't work a full review into my schedule today, but this caught my eye - are there any unintended consequences of that? We don't have a proper CORS policy in place and it's mitigated by a trick. The data layer creates an iframe with `src="https://www.wordpress.com/wp-admin/<some-stuff>/rest-proxy.php` and it's the page embedded in that iframe that makes all the requests which makes them technically same-origin requests. Then, the data layer talks to that iframe via `window.postMessage`. So no direct POST request would really work. I wonder if there are any limitations to GET requests though. Is this direct request for example triggering a preflight OPTIONS request? Would it break if `Access-Control` headers on `public-api` are changed one day? Do you think it's likely it would get affected if we update CSP at some point?

Reply:
Thanks @adamziel 

All this makes sense. I'd probably migrate to redux/data layer for the very reasons you mentioned. 

What I don't want to do is pollute the state too much with a bunch of results (the least of our problems), but we can find a nice compromise.

I should have stated it more clearly in the description, I'm just trying to improve the user experience for now. This is a bashed-out version to get the interaction right.